content/oci/oci.go

```diff
@@ -143,11 +158,56 @@ func (s *Store) Exists(ctx context.Context, target ocispec.Descriptor) (bool, er
 
 // Delete deletes the content matching the descriptor from the store. Delete may
 // fail on certain systems (i.e. NTFS), if there is a process (i.e. an unclosed
-// Reader) using target.
+// Reader) using target. If s.AutoGC is set to true, Delete will recursively
+// remove the dangling nodes caused by the current delete. If s.AutoRemoveReferrers
+// is set to true, Delete will recursively remove the referrers of the manifests
+// being deleted.
 func (s *Store) Delete(ctx context.Context, target ocispec.Descriptor) error {
-	s.sync.Lock()
-	defer s.sync.Unlock()
+	var deleteQueue []ocispec.Descriptor
+	deleteQueue = append(deleteQueue, target)
```

Comment:
nit: We may need a `Queue` struct in `container/queue` package for better readability.